python/imageutils.py

```diff
@@ -0,0 +1,481 @@
+                                                an HDFS URL or an absolute/relative path with the '_HDFSUTILS_FLAG_' flag as prefix, or else it will
+                                                conclude that file is located in local filesystem.
+        in_hdfs                         : boolean variable indicating if image is located in HDFS or local filesystem. By default is set to None.
+                                          If not set (i.e. None), the function will attempt to determine where the file is located.   
+
+        """    
+   
+        if self.utils is None:
+            in_hdfs = False 
+        elif in_hdfs is None:
+            in_hdfs = self.utils.is_hdfs_uri(filepath)
+
+        if in_hdfs:
+            fh = None
+            #gets hdfs path in the case an hdfs uri was provided
+            filepath = self.utils.hdfs_path(filepath)
+
+            with self.utils.client.read(filepath) as reader:
+                stream = reader.read()
+                if self.is_gzipped(filepath, stream[:2]):
+                    fh = nib.FileHolder(fileobj=GzipFile(fileobj=BytesIO(stream)))
+                else:
+                    fh = nib.FileHolder(fileobj=BytesIO(stream))
+                
+                if is_nifti(filename):
+                    return nib.Nifti1Image.from_file_map({'header':fh, 'image':fh})
+                if is_minc(filename):
+                    return nib.Minc1Image.from_file_map({'header':fh, 'image':fh})
+                else:
+                    print('Error: currently unsupported file-format')
+                    sys.exit(1)
+
+        #image is located in local filesystem
+        return nib.load(filepath)
+
+    def is_gzipped(self, filepath, buff=None):
+        
+        """Determine if image is gzipped
+
+
+        Keyword arguments:
+        filepath                        : the absolute or relative filepath to the image
+        buffer                          : the bystream buffer. By default the value is None. If the image is located on HDFS,
+                                          it is necessary to provide a buffer, otherwise, the program will terminate.
+        """
+        mime = magic.Magic(mime=True)
+        try:
+            if buff is None:
+                if 'gzip' in mime.from_file(filepath):
+                    return True
+                return False
+            else:
+                if 'gzip' in mime.from_buffer(buff):
+                    return True
+                return False
+        except:
+            print('ERROR occured while attempting to determine if file is gzipped')
+            sys.exit(1)
+
+    def is_nifti(self):
+        if '.nii' in self.extension:
+            return True
+        return False
+
+
+    def is_minc(self):
+        if '.mnc' in self.extension:
+            return True
+        return False
+
+    
+def get_bytes_per_voxel(dtype):
```

Comment:
For a more pythonic way of implementing a "switch" statement, see https://bytebaker.com/2008/11/03/switch-case-statement-in-python